Build failure: siege

###Steps To Reproduce
#Steps to reproduce the behavior:

```Shell
nix develop
```

### Build log

```
direnv: loading ~/messaging-dispatcher-server/.envrc
direnv: using flake
warning: Git tree '/Users/jayyy/messaging-dispatcher-server' is dirty
warning: Ignoring setting 'auto-allocate-uids' because experimental feature 'auto-allocate-uids' is not enabled
warning: Ignoring setting 'impure-env' because experimental feature 'configurable-impure-env' is not enabled
[1/0/6 built] building siege-4.1.6 (configurePhase): checking for suffix of executables...direnv: ([/opt/homebrew/bin/direnv export zsh]) is taking a while to execute. Use CTRL-C to give up.
error: builder for '/nix/store/aqc9xxdl862253qx2gs6vz9yaawpnl0l-siege-4.1.6.drv' failed with exit code 2;
       last 25 log lines:
       > ./setup.h:99:5: warning: a function declaration without a prototype is deprecated in all versions of C and is treated as a zero-parameter prototype in C2x, conflicting with a previous declaration [-Wdeprecated-non-prototype]
       > int strcasecmp();
       >     ^
       > 1 warning generated.
       > clang -DHAVE_CONFIG_H -I. -I../include   -I/nix/store/f2h8rd0ma4ml9hrqqdvdj5vhxilx1pfl-zlib-1.3-dev/include/zlib -I/nix/store/f2h8rd0ma4ml9hrqqdvdj5vhxilx1pfl-zlib-1.3-dev/include -I/nix/store/rl8550bfqka2y6fz61xvjki9yxdfvq74-openssl-3.0.12-dev/include/openssl -I/nix/store/rl8550bfqka2y6fz61xvjki9yxdfvq74-openssl-3.0.12-dev/include -I/nix/store/f2h8rd0ma4ml9hrqqdvdj5vhxilx1pfl-zlib-1.3-dev/include/zlib -I/nix/store/f2h8rd0ma4ml9hrqqdvdj5vhxilx1pfl-zlib-1.3-dev/include -D_REENTRANT -D_POSIX_PTHREAD_SEMANTICS -W -Wall -Wunused-value -DOPENSSL_NO_KRB5  -g -O2 -c cookies.c
       > In file included from cookies.c:12:
       > ./setup.h:99:5: warning: a function declaration without a prototype is deprecated in all versions of C and is treated as a zero-parameter prototype in C2x, conflicting with a previous declaration [-Wdeprecated-non-prototype]
       > int strcasecmp();
       >     ^
       > cookies.c:69:11: error: incompatible pointer to integer conversion initializing 'size_t' (aka 'unsigned long') with an expression of type 'pthread_t _Nonnull' (aka 'struct _opaque_pthread_t *') [-Wint-conversion]
       >   size_t  id    = pthread_self();
       >           ^       ~~~~~~~~~~~~~~
       > cookies.c:120:23: warning: comparison between pointer and integer ('size_t' (aka 'unsigned long') and 'pthread_t' (aka 'struct _opaque_pthread_t *')) [-Wpointer-integer-compare]
       >     if (cur->threadID == id) {
       >         ~~~~~~~~~~~~~ ^  ~~
       > cookies.c:148:23: warning: comparison between pointer and integer ('size_t' (aka 'unsigned long') and 'pthread_t' (aka 'struct _opaque_pthread_t *')) [-Wpointer-integer-compare]
       >     if (cur->threadID == id) {
       >         ~~~~~~~~~~~~~ ^  ~~
       > cookies.c:172:10: error: incompatible pointer to integer conversion initializing 'size_t' (aka 'unsigned long') with an expression of type 'pthread_t _Nonnull' (aka 'struct _opaque_pthread_t *') [-Wint-conversion]
       >   size_t id = pthread_self();
       >          ^    ~~~~~~~~~~~~~~
       > 3 warnings and 2 errors generated.
       > make[1]: *** [Makefile:439: cookies.o] Error 1
```

### Notify maintainers
@7c6f434c

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"aarch64-darwin"`
 - host os: `Darwin 23.3.0, macOS 14.3`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.20.1`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/si/siege/package.nix
+++ b/pkgs/by-name/si/siege/package.nix
@@ -33,6 +33,9 @@ stdenv.mkDerivation rec {
     "--with-zlib=${zlib.dev}"
   ];
 
+  # Borrowed solution from homebrew: https://github.com/Homebrew/homebrew-core/blob/1c7c95183c0984a84b1680422afab6578c300a27/Formula/s/siege.rb#L31
+  CFLAGS = lib.optionalString stdenv.cc.isClang "-Wno-int-conversion";
+
   meta = {
     description = "HTTP load tester";
     homepage = "https://www.joedog.org/siege-home/";

